projects/core/src/occ/services/occ-endpoints.service.ts

```diff
@@ -58,52 +87,148 @@ export class OccEndpointsService {
    * @param endpoint Endpoint suffix
    */
   getOccEndpoint(endpoint: string): string {
-    if (!this.config?.backend?.occ) {
-      return '';
-    }
-    endpoint = this.config.backend.occ.endpoints?.[endpoint];
+    endpoint = this.getRawEndpointValue(endpoint);
+
+    return this.getEndpoint(endpoint, { baseSite: false });
+  }
 
+  /**
+   * @Deprecated since 3.2 - use "buildOccBaseUrl" with the same parameters
+   *
+   * Returns base OCC endpoint (baseUrl + prefix + baseSite) by if no parameters are specified
+   *
+   * @param propertiesToOmit Specify properties to not add to the url (baseUrl, prefix, baseSite)
+   */
+  getBaseEndpoint(propertiesToOmit?: BaseOccUrlProperties): string {
+    return this.getOccBaseUrl(propertiesToOmit);
+  }
+
+  /**
+   * @Deprecated since 3.2 - use "buildUrlFromEndpointString" with the same parameters
+   *
+   * Returns an OCC endpoint including baseUrl and baseSite
+   *
+   * @param endpoint Endpoint suffix
+   * @param propertiesToOmit Specify properties to not add to the url (baseUrl, prefix, baseSite)
+   */
+  getEndpoint(
+    endpoint: string,
+    propertiesToOmit?: BaseOccUrlProperties
+  ): string {
     if (
       !endpoint.startsWith('/') &&
       !this.config.backend.occ.prefix.endsWith('/')
     ) {
       endpoint = '/' + endpoint;
     }
 
-    return (
-      this.config.backend.occ.baseUrl +
-      this.config.backend.occ.prefix +
-      endpoint
-    );
+    return this.buildUrlFromEndpointString(endpoint, propertiesToOmit);
   }
 
   /**
-   * Returns base OCC endpoint (baseUrl + prefix + baseSite)
+   * Returns base OCC endpoint (baseUrl + prefix + baseSite) base on provided values
+   *
+   * @param propertiesToOmit Specify properties to not add to the url (baseUrl, prefix, baseSite)
    */
-  getBaseEndpoint(): string {
+  getOccBaseUrl(propertiesToOmit?: BaseOccUrlProperties): string {
     if (!this.config?.backend?.occ) {
       return '';
     }
 
-    return (
-      (this.config.backend.occ.baseUrl || '') +
-      this.config.backend.occ.prefix +
-      this.activeBaseSite
-    );
+    if (!propertiesToOmit) {
+      return urlPathJoin(
+        this.config.backend.occ.baseUrl || '',
```

Comment:
The config handles undefined values right?

```suggestion
        this.config.backend.occ.baseUrl,
```

Reply:
Yes it does